torch/csrc/jit/script/init.cpp

```diff
@@ -376,6 +352,27 @@ py::object runMethodFromPython(Method& m, py::args args) {
   return wrapStack(std::move(stack));
 }
 
+Resolver pythonResolver(ResolutionCallback rcb) {
+  return [=](const std::string& name) -> std::shared_ptr<SugaredValue> {
+      AutoGIL ag;
+      py::object obj = rcb(name);
+      if(obj.is(py::none())) {
+        return nullptr;
+      }
+      // Inline *ONLY PURE* ScriptModules. This allows us to call arbitrary
+      // @script functions within a scripting context.
+      if (py::isinstance<Module>(obj)) {
```

Comment:
This will cause regressions. Python resolver is only called for global variables. However PythonValues can be derived in other ways (e.g. a module `a.b`). In these cases you are missing the check.